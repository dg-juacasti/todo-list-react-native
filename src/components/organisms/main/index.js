import React, { useState, useEffect } from "react";
import {StyledMain, StyledSearchContent, StyledContent} from './index.styles';
import {Typography} from '../../atoms/typography';
import {CustomButton} from '../../atoms/button';
import {TasksList} from '../../molecules/tasksList';
import { COLORS } from "../../../shared/colors";
import fonts from "../../../shared/fonts";
import {useTasksList} from '../../../hooks/useTasksList';
import { Input } from "../../atoms/input";


export function Main({ navigation }) {
  const {tasks, getTasks} = useTasksList();
  const [saveTasks, setSaveTasks] = useState([]);
  const [descriptionToSearch, setDescriptionToSearch] = useState('');

  useEffect(() => {
    getTasks();
  }, []);

  useEffect(() => {
    setSaveTasks(tasks);
  }, [tasks]);

  useEffect(() => {
    navigation.addListener('focus', () => {
      getTasks();
    });
  }, []);
  
  const onRefresh = () => {
    getTasks();
  }

  return (
    <StyledMain>
      <StyledContent>
        <Typography
          caption="Todo List"
          color={COLORS.textColor}
          textAlign="center"
          fontFamily={fonts.RobotoBold}
          fontSize={24}
        />
      </StyledContent>
      <StyledSearchContent>
        <Input
          testID="searchId"
          value={descriptionToSearch}
          placeholder="Buscar tarea"
          onChangeText={(value) => setDescriptionToSearch(value)}
          borderColor={COLORS.textColor1}
          borderWidth={1}
          borderRadius={5}
          height={40}
          width={280}
        />
        <CustomButton
          onPress={() => navigation.navigate('Form')}
          disabled={false}
          backgroundColor={COLORS.textColor}
          height={40}
          width={40}
          iconName="plus-circle"
          iconColor={COLORS.white}
        />
      </StyledSearchContent>
      <TasksList 
        tasks={saveTasks}
        descriptionToSearch={descriptionToSearch}
        onRefresh={onRefresh}
        navigation={navigation} />
    </StyledMain>
  );
}