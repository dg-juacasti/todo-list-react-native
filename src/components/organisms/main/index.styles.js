import styled from 'styled-components/native';


export const StyledMain = styled.View`
  padding: 24px;
`;

export const StyledContent = styled.View`
  margin-top: 24px;
`;
export const StyledSearchContent = styled.View`
  flex-direction: row;
  justify-content: space-between;
  margin-top: 24px;
  margin-bottom: 16px;
`;
