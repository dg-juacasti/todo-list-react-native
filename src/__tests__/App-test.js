import React from "react";
import { render, fireEvent, screen, act } from "@testing-library/react-native";
import axios from "axios";
import { Main } from '../components/organisms/main';

describe("TodoList App tests", () => {
  it("Should render a list with several todos", async () => {
    const component = render(<Main />);
    expect(component).toMatchSnapshot();
  });

  it("Should call search input", async () => {
    const descriptionToSearch = 'Otra tarea';

    const { getByTestId } = render(<Main />);
    const input = getByTestId('searchId');
    fireEvent.changeText(input, descriptionToSearch);
    expect(input.props.value).toBe(descriptionToSearch);
  });

  it("Should create a new todo", async () => {});

  /**
   *
   * Probar que el formulario muestre los mensajes de requerimiento cuando
   * el formulario no tenga la descripción y la fecha ingresada
   */
  it("Should validate the todo form, description and date required", async () => {});

  it("Should update a todo, description and date", async () => {});

  it("Should delete a todo", async () => {});

  /**
   *
   * Probar que el formulario muestre los mensajes de requerimiento cuando
   * el formulario no tenga la descripción y la fecha ingresada
   */
  it("Should update the todo status ", async () => {});

  it("Should show an message when  the todo list is empty  ", async () => {});

  /**
   *
   * Probar que la barra de estado cambia cuando se completa una tarea
   * se puede probar por el cambio en texto o por porcentaje de completitud
   */
  it("Should the progress bar change its label text or percentage when a todo is completed ", async () => {});

  /**
   *
   * Probar el filtro de las tareas por descripcion
   */
  it("Should filter the todo list by description", async () => {});

  /**
   *
   * Probar el filtro de tareas que falta por completar y que una vez esten filtradas a dar click nuevamente
   * sobre el boton del filtro se muestren todos la lista nuevamente
   */
  it("Should filter the todo list by completed status and toggle functionality button", async () => {});
});
