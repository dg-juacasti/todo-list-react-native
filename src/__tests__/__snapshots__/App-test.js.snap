// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList App tests Should render a list with several todos 1`] = `
<View
  style={
    Array [
      Object {
        "paddingBottom": 24,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 24,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginTop": 24,
        },
      ]
    }
  >
    <Text
      color="#1E6395"
      fontFamily="Roboto-Bold"
      fontSize={24}
      style={
        Array [
          Object {
            "alignItems": "center",
            "color": "#1E6395",
            "fontFamily": "Roboto-Bold",
            "fontSize": 24,
            "fontWeight": "normal",
            "justifyContent": "center",
            "textAlign": "center",
          },
        ]
      }
      textAlign="center"
    >
      Todo List
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 16,
          "marginTop": 24,
        },
      ]
    }
  >
    <TextInput
      borderColor="#85A2C2"
      borderRadius={5}
      borderWidth={1}
      height={40}
      onChangeText={[Function]}
      placeholder="Buscar tarea"
      testID="searchId"
      value=""
      width={280}
    />
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1E6395",
          "borderRadius": 3,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 40,
        }
      }
    >
      <Text
        color="#ffffff"
        style={
          Array [
            Object {
              "alignItems": "center",
              "color": "#ffffff",
              "fontFamily": "Roboto-Regular",
              "fontSize": 16,
              "fontWeight": "normal",
              "justifyContent": "center",
              "textAlign": "left",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "marginLeft": undefined,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 25,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ÔÅï
        </Text>
      </View>
    </View>
  </View>
  <View />
</View>
`;
